proposals/distribution.md

```diff
@@ -1,19 +1,25 @@
 # Abstract
 
-The Docker registry protocol has become the defacto standard across the container registry world ([https://github.com/docker/distribution/blob/master/docs/spec/api.md](https://github.com/docker/distribution/blob/master/docs/spec/api.md)).
+The Docker registry protocol has become the defacto standard across the container registry world.
 
 In the OCI, having a solid, common distribution specification with conformance testing will ensure long lasting security and interoperability throughout the container ecosystem.
 
 ## Proposal
 
-TL;DR; Move [https://github.com/docker/distribution/tree/master/docs/spec](https://github.com/docker/distribution/tree/master/docs/spec) to [https://github.com/opencontainers/distribution-spec](https://github.com/opencontainers/distribution-spec)
+TL;DR; Move [`api.md`][api.md] to a new [distribution-spec project](https://github.com/opencontainers/distribution-spec).
 
 This proposal covers the distribution API spec, and while it does not cover the code for the docker-registry, that implementation is considered the reference implementation. There are other implementations of this protocol, not all are open-source though (Google gcr.io, Amazon ECR, CoreOS Quay, Gitlab registry, JFrog Artifactory registry, Huawei Dockyard, etc).
 
 In the past when the topic of having an OCI specification around the distribution of container images was discussed, it was deferred as "let’s get the image format defined, meanwhile the industry will settle on a distribution standard". Fast forward, OCI image format is out and adopted, and the Registry v2 is the defacto standard. There is and will be use-cases for alternate methods and the future will likely hold creative ways to push, fetch and share container images, but right now this promotion serves to acknowledge by the OCI the current industry standard of distributing container images.
 
 There is polish that is needed e.g. broken links to storage-driver docs, as well as making sections more generic regarding the OCI descriptors and media-types, but on the whole this is a lateral move.
 
+This project is scoped to cover the client ↔ registry interaction.  The following are out of scope:
+
+* Authentication and authorization.  Docker's current registry uses an [extension][token] of the [`Bearer`][rfc6750] [auth scheme][rfc7235-s2.1].  Work on specifying Docker's scheme will continue independently, and is orthogonal to the registry API.
```

Comment:
Is there anything today which you see as incompatible with standard [bearer](https://tools.ietf.org/html/rfc6750) or [basic auth](https://tools.ietf.org/html/rfc2617). I think this needs to highlight that bearer and basic auth are in scope as supported mechanisms, however, the extra fields, interpretation of the specified fields, and method for getting a bearer token are not currently in scope.

Reply:
> Is there anything today which you see as incompatible with standard bearer or basic auth.

Nothing in the API spec; it can support anything via the vanilla [`WWW-Authenticate`][1] header.  And that would work perfectly for [`Basic`][2] auth or most other [auth schemes][3] out of the box.  The issue I see with [Docker's `Bearer` extentions][4] is just that they're extensions.  Someone getting a `WWW-Authenticate: Bearer …` response and looking [in the IANA registry][3] wouldn't be able to discover the Docker extensions (e.g. the fact that `realm` is the recommended auth-server).  And it's not obvious from the `WWW-Authenticate` header alone whether the server intended [vanilla `Bearer` semantics][5] or the Docker extension semantics.

> I think this needs to highlight that bearer and basic auth are in scope as supported mechanisms…

I don't think you need to do that with normative language, and this scope restriction does not restrict your ability to add informative notes.  So you can certainly mention that many registries will use [`Bearer`][5].  But a full decoupling would mean that the only *normative* link would be via [`WWW-Authenticate`][1] and the [IANA scheme list][3].

[1]: https://tools.ietf.org/html/rfc7235#section-4.1
[2]: https://tools.ietf.org/html/rfc7617
[3]: http://www.iana.org/assignments/http-authschemes/http-authschemes.xhtml
[4]: https://github.com/docker/distribution/blob/5cb406d511b7b9163bff9b6439072e4892e5ae3b/docs/spec/auth/token.md
[5]: https://tools.ietf.org/html/rfc6750